src/Targets/Microsoft.Managed.DesignTime.targets

```diff
@@ -283,4 +278,15 @@
   <!-- This target collects all the resolved references that are used to actually compile. -->
   <Target Name="CollectResolvedCompilationReferencesDesignTime" DependsOnTargets="CompileDesignTime" Returns="@(ReferencePathWithRefAssemblies)" />
 
+  <!-- This target collects all up-to-date file additions. -->
+  <Target Name="CollectUpToDateInputsDesignTime" DependsOnTargets="CompileDesignTime" Returns="@(UpToDateCheckInput)">
+    <!-- If the project we're compiling has a ruleset, make sure we check it for up-to-date checks. -->
+    <ItemGroup Condition="'$(ResolvedCodeAnalysisRuleSet)' != ''">
+      <UpToDateCheckInput Include="$([System.IO.Path]::GetFullPath($(ResolvedCodeAnalysisRuleSet)))" />
```

Comment:
This seems a very expensive way of getting the resolving the ruleset - we do a design-time build to figure if we need to do a full build?

Is there a reason you went down this approach instead of just rooting UpToDateCheckInput/Output against the project file in the rule data?